typescript/sdk/src/crud/EvmHookModule.ts

```diff
@@ -13,32 +14,29 @@ import { CrudModule, CrudModuleArgs } from './AbstractCrudModule.js';
 export class EvmHookModule extends CrudModule<
   ProtocolType.Ethereum,
   HookConfig,
-  HyperlaneAddresses<HookFactories>
+  HyperlaneAddresses<HookFactories> & {
+    deployedHook: Address;
+  }
 > {
   protected logger = rootLogger.child({ module: 'EvmHookModule' });
   protected reader: EvmHookReader;
 
   protected constructor(
     protected readonly multiProvider: MultiProvider,
```

Comment:
Sorry I should have raised this concern earlier but was preoccupied with all the refactoring work I was juggling but all new SDK tooling should really be multi-protocol first, otherwise we're going to have to re-write it as we've done other classes in the past. For example, we should be using a MultiProtocolProvider.

If this requires more discussion or is a big change, feel free to handle in a separate PR.